File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n=s.size();
        for(int i=0;i<n;++i){
            bool found=false;
            for(int j=0;j<n;++j){
                if(i!=j && s[i]==s[j]){
                    found=true;
                    break;
                }
            }
            if(!found){
                return i;
            }
        }
        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
# class Solution(object):
#     def firstUniqChar(self, s):
#         for i in range(len(s)):
#             if s.count(s[i])==1:
#                 return i
#         return -1


class Solution(object):
    def firstUniqChar(self, s):
        freq={}
        for ch in s:
            if ch in freq:
                freq[ch]+=1
            else:
                freq[ch]=1
        for i in range(len(s)):
            if freq[s[i]]==1:
                return i
        return -1
